R/prepare_predictions.R

```diff
@@ -107,9 +107,13 @@ prepare_predictions.brmsterms <- function(x, draws, sdata, data, ...) {
   for (dp in valid_dpars) {
     dp_regex <- paste0("^", dp, resp, "$")
     if (is.btl(x$dpars[[dp]]) || is.btnl(x$dpars[[dp]])) {
+      old_acdata <- NULL
+      if (parameterize_ac_effects(x$dpars[[dp]])) {
```

Comment:
why do we need to rerun full `make_standata` here again? in prepare predictions, we already compute `standata` and pass is as variable `sdata`.

Reply:
The reason for this is that we need some information on the old grouping levels (which isn't present in `sdata` since that is computed using the `newdata`) to make predictions conditional on the original data. However, perhaps it's better to split out a helper function similar to `prepare_predictions_ranef` that packages only the necessary metadata and posterior draws for the predictions requested.

Reply:
I see. brms has it's own process in place for this purpose. See `standata_basis` in `make_standata.R`